core/templates/pages/practice-session-page/practice-session-page.component.ts

```diff
@@ -16,127 +16,136 @@
-          UrlService.getTopicUrlFragmentFromLearnerUrl());
-        var practiceSessionsDataUrl = UrlInterpolationService
-          .interpolateUrl(
-            PRACTICE_SESSIONS_DATA_URL, {
-              topic_url_fragment: topicUrlFragment,
-              classroom_url_fragment: (
-                UrlService.getClassroomUrlFragmentFromLearnerUrl()),
-              stringified_subtopic_ids: ctrl.stringifiedSubtopicIds
-            });
-        var practiceSessionsUrl = UrlInterpolationService.interpolateUrl(
-          PRACTICE_SESSIONS_URL, {
-            topic_url_fragment: topicUrlFragment,
-            classroom_url_fragment: (
-              UrlService.getClassroomUrlFragmentFromLearnerUrl()),
-            stringified_subtopic_ids: ctrl.stringifiedSubtopicIds
-          });
-        var topicViewerUrl = UrlInterpolationService.interpolateUrl(
-          TOPIC_VIEWER_PAGE, {
-            topic_url_fragment: topicUrlFragment,
-            classroom_url_fragment: (
-              UrlService.getClassroomUrlFragmentFromLearnerUrl()),
-          });
-        $http.get(practiceSessionsDataUrl).then(function(result) {
-          var skillList = [];
-          var skillDescriptions = [];
-          for (var skillId in result.data.skill_ids_to_descriptions_map) {
-            skillList.push(skillId);
-            skillDescriptions.push(
-              result.data.skill_ids_to_descriptions_map[skillId]);
-          }
-          var questionPlayerConfig = {
-            resultActionButtons: [
-              {
-                type: 'REVIEW_LOWEST_SCORED_SKILL',
-                i18nId: 'I18N_QUESTION_PLAYER_REVIEW_LOWEST_SCORED_SKILL'
-              },
-              {
-                type: 'DASHBOARD',
-                i18nId: 'I18N_QUESTION_PLAYER_MY_DASHBOARD',
-                url: topicViewerUrl
-              },
-              {
-                type: 'RETRY_SESSION',
-                i18nId: 'I18N_QUESTION_PLAYER_NEW_SESSION',
-                url: practiceSessionsUrl
-              },
-            ],
-            skillList: skillList,
-            skillDescriptions: skillDescriptions,
-            questionCount: TOTAL_QUESTIONS,
-            questionsSortedByDifficulty: false
-          };
-          ctrl.questionPlayerConfig = questionPlayerConfig;
-          ctrl.topicName = result.data.topic_name;
-          ctrl.setPageTitle();
-          ctrl.subscribeToOnLanguageCodeChange();
-          LoaderService.hideLoadingScreen();
+    const translatedTitle = this.translateService.instant(
+      'I18N_PRACTICE_SESSION_PAGE_TITLE', {
+        topicName: this.topicName
+      }
+    );
+
+    this.pageTitleService.setDocumentTitle(translatedTitle);
+  }
+
+  subscribeToOnLanguageCodeChange(): void {
+    this.directiveSubscriptions.add(
+      this.i18nLanguageCodeService.onI18nLanguageCodeChange.subscribe(() => {
+        this.setPageTitle();
+      })
+    );
+  }
+
+  _fetchSkillDetails(): void {
+    var topicUrlFragment = (
```

Comment:
prefer `let/const` over `var` here and elsewhere.

Reply:
done